hot-supabase/contributors.js

```diff
@@ -0,0 +1,118 @@
+import consoleHeader from './lib/consoleHeader.js'
+import api from "./lib/persistedGraphQL.js";
+import fetchContributorNames from "./lib/contributorNameHelper.js";
+import {p} from "@antfu/utils";
+import {supabase} from "./lib/supabase.js";
+
+const limitDays = parseInt(process.env.LIMIT_CONTRIBUTOR_DAYS) || 5
+const lastExecuted = new Date()
+
+consoleHeader(' OPEN |SAUCED', {
+  font: 'block',
+})
+console.log(`Started execution at ${lastExecuted}`)
+
+async function run() {
+  const testDate = new Date(lastExecuted.getTime() - (limitDays * 24 * 60 * 60 * 1000))
+
+  const {data: getAvailableRepos, error} = await supabase
+    .from('repos')
+    .select('*')
+    .lte('last_fetched_contributors_at', JSON.stringify(testDate))
+    .limit(process.env.LIMIT_CONTRIBUTOR_REPOS)
+
+  if (error) {
+    console.log(`Unable to fetch repos from supabase`, error)
+    return
+  }
+
+  const contributions = []
+
+  await p(getAvailableRepos)
```

Comment:
this could run outside of the loop but more efficient might mean more rate limiting